keyboards/install.php

```diff
@@ -0,0 +1,266 @@
+        if(!empty(self::$bcp47)) $url .= "&bcp47=".rawurlencode(self::$bcp47);
+        $url = htmlspecialchars($url);
+        return <<<END
+        <div class='download download-$platform'>
+
+        <ol>
+          <li id='step1'><a href='$keymanUrl' title='Download and install Keyman'>Install $keymanProductName</a></li>
+          <li id='step2'><a class='download-link binary-download' href='$url' onclick='return downloadBinaryFile(this);'><span>Install $title</span></a></li>
+        </ol>
+
+        </div>
+END;
+      } else {
+        // TODO: this message not yet clear
+        return "<div class='download download-platform'>Not available for $platform</div>";
+      }
+    }
+
+    protected static function WriteWindowsBoxes() {
+      $keyboard = self::$keyboard;
+      $bcp47 = rawurlencode(empty(self::$bcp47) ? '' : self::BOOTSTRAP_SEPARATOR.self::$bcp47);
+      $tier = self::$tier;
+      $version = self::$versions->windows->$tier;
+      $e_keyboard_id = rawurlencode($keyboard->id);
+      $h_keyboard_name = htmlentities($keyboard->name);
+
+      // Note, we don't need to capture an event for the keyboard download here, because we'll capture it when the bootstrap downloads the file.
+      $downloadLink = KeymanHosts::Instance()->downloads_keyman_com . "/windows/{$tier}/{$version}/keyman-setup" . self::BOOTSTRAP_SEPARATOR . "{$keyboard->id}{$bcp47}.exe";
+      $downloadLinkE = json_encode($downloadLink, JSON_UNESCAPED_SLASHES);
+      $helpLink = KeymanHosts::Instance()->help_keyman_com . "/products/desktop/current-version/docs/start_download-install_keyman";
+
+      $keyboardHomeUrl = "/keyboards/{$e_keyboard_id}";
+      if(!empty(self::$bcp47)) $keyboardHomeUrl .= "?bcp47=" . rawurlencode(self::$bcp47);
+
+      $result = <<<END
+<div class='download download-windows'>
+<p>Your $h_keyboard_name keyboard download should start shortly. If it does not, <a href='$downloadLink'>click here</a> to start the download.</p>
+<script>
+  window.setTimeout(function() {
+    if(document.documentElement.getAttribute('data-platform') == 'windows') {
+      // TODO: show arrow in window where download is likely to be accessible from; this is browser+browser version dependent :(
+      // TODO: don't forget to look for a library that implements this which may save pain
+      location.href = $downloadLinkE;
+    }
+  }, 1000);
+</script>
+<ul>
+<li><a href='$helpLink'>Help on installing Keyman</a></li>
+<li><a href='$keyboardHomeUrl'>{$h_keyboard_name} keyboard home</a></li>
+</ul>
+</div>
+
+END;
+      return $result;
+    }
+
+    protected static function WriteMacOSBoxes() {
+      return self::download_box(
+        'Keyman for macOS',
+        KeymanDownloadVersions::getDownloadUrl('mac'), // note inconsistent platform name :(
+        htmlentities(self::$keyboard->name) . ' for macOS',
+        'Installs only ' . htmlentities(self::$keyboard->name) . '. <a href="/macosx">Keyman for Mac</a> must be installed first.',
```

Comment:
```suggestion
        'Installs only ' . htmlentities(self::$keyboard->name) . '. <a href="/mac">Keyman for macOS</a> must be installed first.',
```

Reply:
The macOS part needs to be rewritten anyway -- that'll come as I do Universal Link work for mac. For now, only the Windows code is complete.